base/iobuffer.jl

```diff
@@ -101,8 +106,9 @@ function IOBuffer(;
         write::Union{Bool,Nothing}=true,
         append::Union{Bool,Nothing}=nothing,
         truncate::Union{Bool,Nothing}=true,
-        maxsize::Integer=typemax(Int))
-    size = maxsize == typemax(Int) ? 32 : Int(maxsize)
+        maxsize::Integer=typemax(Int),
+        sizehint::Union{Integer,Nothing}=nothing)
+    size = maxsize != typemax(Int) ? Int(maxsize) : sizehint !== nothing ? Int(sizehint) : 32
```

Comment:
It seems like `sizehint` should take precedence over `maxsize` here.

Reply:
e.g. consider a case where `maxsize` is 2GB (because the string will be sent someplace 32-bit) but `sizehint` is 8.   We probably don't want to allocate 2GB.

Reply:
I forgot to fix that. Now fixed. Thank you.